src/app.js

```diff
@@ -1,12 +1,88 @@
+import axios from "axios";
+import distance from "euclidean-distance";
+
+// Get the API token
+async function getToken() {
+  let token;
+  try {
+    const response = await axios.post(
+      "https://blue-bottle-api-test.herokuapp.com/v1/tokens",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    token = response.data.token;
+  } catch (e) {
+    console.error(`Exception occurred while fetching the token: ${e}`);
+  }
+  return token;
+}
+
+// Get all shop data from the API
+async function getShopData() {
+  const token = await getToken();
+
+  try {
+    const response = await axios.get(
+      `https://blue-bottle-api-test.herokuapp.com/v1/coffee_shops?token=${token}`,
+      {
+        headers: {
+          Accept: "application/json",
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    if (e.response.status == 401 || 406 || 503 || 504) {
+      console.error(`Exception occurred while fetching shop list : ${e}`);
+      return [];
+    }
+  }
+}
+
 /**
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
```

Comment:
I like that you used Javadocs for documentation, getting IDE suggestions and help is greatly appreciated. :rocket: 
One question I have for you regarding this: would you write documentation like this for every function? Where do you draw the line and stop writing documentation?
Also, what do you think is the difference between documentation using Javadocs, and plain comments?

Reply:
In fact, Javadocs for the function came from the original repository so I did not write it :) I know that javadocs can be used to generate automated documentation for projects and they also help developers to understand the code better. It also defines a basic contract for what the code should take as arguments or return at the end. My javadocs knowledge is very limited right now. But I am eager to learn more about it and use it in every piece of code I write.